packages/dds/map/src/index.ts

```diff
@@ -50,41 +50,20 @@ import type {
 	IChannelFactory,
 	IFluidDataStoreRuntime,
 } from "@fluidframework/datastore-definitions";
+import type { ISharedObjectKind } from "@fluidframework/shared-object-base";
 import { MapFactory } from "./map.js";
 import { ISharedMap } from "./interfaces.js";
 
 /**
  * {@inheritDoc ISharedMap}
  * @public
- * @deprecated Please use SharedTree for new containers.  SharedMap is supported for loading preexisting Fluid Framework 1.0 containers only.
+ * @deprecated Please use SharedTree for new containers. SharedMap is supported for loading preexisting Fluid Framework 1.0 containers only.
  */
-export const SharedMap = {
-	/**
-	 * Get a factory for SharedMap to register with the data store.
-	 * @returns A factory that creates SharedMaps and loads them from storage.
-	 */
+export const SharedMap: ISharedObjectKind<ISharedMap> = {
```

Comment:
Can we mimic `SharedObjectClass` here so our users don't get confused?

Reply:
This change deletes SharedObjectClass (which no longer had anything to do with classes so its name seems poor) and replaces it with ISharedObjectKind, which this code can and explicitly refer to (unlike SharedObjectClass due to the package it was in) by using it as the type. This means that the doc comments from ISharedObjectKind explain its members unlike with SharedObjectClass.

I'm unclear on what you would like done differently: I feel like this whole PR is trying to do what you are requesting here by providing explicit types and docs to connect this code to the container schema stuff to make it more clear to users (and developers).